### ISSUE TITLE
[c++] virtual and related functions not supported

### ISSUE THREAD

**Issue by bartlibert** [kkoomen/vim-doge#41]
**Describe the bug**
In c++, function can be marked virtual. DoGe does not support them. When the "virtual" keyword is removed, everything works.
Also, a virtual function can be marked as "pure virtual" by appending "=0" after it. This is not supported either.
Finally, when a child class overrides a virtual function from its parent, it can/must be marked "override" or "final", this is not supported either.

**To Reproduce**
1. Create a file test.h with the following content:
```c++
class Test {
    virtual void my_virtual_function();
    virtual void my_pure_virtual_function() = 0;
};

class Test2: public Test {
    void my_virtual_function() override;
    void my_pure_virtual_function() final;
}
```
2. Try to generate documentation for any of the my_virtual_function of my_pure_virtual_function lines.

**Expected behavior**
Documentation is created.

**Comment by kkoomen** [comment]
We're working on a good and stable implementation for the C-family, but we're a bit stuck at #55 using a python binding of clang. If you have any experience or want to help out, please do. At this moment we're stuck. If we fix this we can probably get a good C version running.

**Comment by kkoomen** [comment]
Development progress for C++ can be tracked via PR #56.

**Comment by kkoomen** [comment]
@bartlibert Can you provide me with clear documentation results that you'd expect in your code? If you have done that I can fix this immediately.

**Comment by kkoomen** [comment]
@bartlibert We're doing a rewrite without clang with a new modern parser called tree-sitter where this issue will be fixed.

**Comment by kkoomen** [comment]
This bug has been fixed and released in [v3.4.0](https://github.com/kkoomen/vim-doge/tree/v3.4.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
